android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -930,6 +943,17 @@ class GeckoEngine(
             unBlockedBySmartBlock = this.blockingData.any { it.unBlockedBySmartBlock() },
         )
     }
+
+    internal fun org.mozilla.geckoview.WebExtension?.toSafeWebExtension(): GeckoWebExtension? {
```

Comment:
Shouldn't we remove this?

Reply:
We removed the [suppression](https://github.com/mozilla-mobile/firefox-android/commit/cbfc8d34e086769d2d8184ec311a1a1dec0ce16e#diff-2ddc504c9aef157255a3ebc15ca5ccadbf966934d724e5a52f8db8b15da4180fR949-R950) and the `this.id != null` check.

```kotlin
    @Suppress("SENSELESS_COMPARISON")
        return if (this != null && this.id != null) {
```

But still we need to verify that the gecko view extension is not null before try to create an AC extension :)